### PR TITLE
3D ItI solver implementation

### PR DESCRIPTION
**Pull request by jma02**
## Summary
This PR adds a complete 3D ItI solver implementation to jaxhps, matching the existing 2D ItI functionality:

- 3D ItI precompute operators (Q_H, G, N matrices) with spectral convergence verification
- Local solve stage with Robin boundary conditions and shape contracts matching 2D ItI
- Eight-way octree merge for ItI maps with spectral convergence on plane-wave tests
- Down-pass and dispatch wiring for integration with build_solver/solve
- Variable-coefficient manufactured-source tests
- Accuracy check examples from upstream author
- Exterior-source ItI test cases

The implementation follows the same conventions as the existing 2D ItI code, with outer boundary using 1st-order Sommerfeld condition (eta = -κ).

#### Test plan
- Run the 3D ItI accuracy checks: `python examples/accuracy_check_3D_ItI_merges.py --problem_1 --problem_2 --p_vals 4 6 8 --l_vals 1 2`
- Run the unit tests for local solve, merge, and pipeline stages
- Verify spectral convergence on plane-wave tests

Generated with [Devin](https://cli.devin.ai/docs)

### CODE REVIEW

File: src/jaxhps/_pdeproblem.py [hunk 1]
```diff
@@ -207,10 +215,26 @@ def __init__(
...
+                self.Q = precompute_Q_3D_DtN(
+                    domain.p, domain.q, self.D_x, self.D_y, self.D_z
+                )
+            else:
+                # Interpolation / Differentiation matrices for 3D ItI merges.
+                # Same naming convention as the 2D case (P, G, QH).  In 3D the
+                # Gauss -> Cheby boundary interpolation `P` is the same for
+                # DtN and ItI (purely geometric averaging at shared
+                # edges/corners), so we reuse the DtN constructor here.
+                self.P = precompute_P_3D_DtN(domain.p, domain.q)
```
Comment (jma02) [inline src/jaxhps/_pdeproblem.py:229]: Let me know if we should lift this function up into a shared function

File: src/jaxhps/down_pass/_uniform_3D_ItI.py [hunk 2]
```diff
@@ -0,0 +1,341 @@
+"""3D ItI down-pass for uniform octrees.
+
```
Comment (jma02) [inline src/jaxhps/down_pass/_uniform_3D_ItI.py:2]: Let me know if the comment density is acceptable
